1주차 과제

ddaeyeonkim opened the pull request:
- 코인 리스트 가져오기

sjjeong committed:
스터디 초반 셋팅 과정 0, 1 번째 추가

sjjeong committed:
커리큐럼 추가

ddaeyeonkim committed:
Merge branch 'master' into class01/improve777

ddaeyeonkim committed:
build.gradle 의존성 추가

ddaeyeonkim committed:
data class 추가

ddaeyeonkim committed:
xml 수정 및 추가

- activity_main
- fragment_ticker
- item_ticker

ddaeyeonkim committed:
통신 모듈 추가

ddaeyeonkim committed:
TickerFragment 추가

ddaeyeonkim committed:
manifest 인터넷 권한 추가

GongSulMin commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/data/source/UpbitRepository.kt:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    override fun getMarkets(success: (markets: String) -> Unit?, fail: (msg: String) -> Unit?) {
+        if (markets == null) {
```
수정할 부분은 아닌데 궁금한 부분이 있어서요!
null 처리 하신게 다시 refresh 하실 때 
market 정보는 새로 안불러오신거죠???

ddaeyeonkim replied:
넵 마켓 정보는 빈번하게 리프레시 될 일이 없을 것 같아서 처음 한번만 부르게 하고 
필드로 저장해서 사용하도록 했습니다

GongSulMin replied:
감사합니당 :)

sjjeong commented on improve777/GAASProject/app/src/main/res/layout/fragment_ticker.xml:
```diff
@@ -0,0 +1,91 @@
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
```
SwipeRefreshLayout 하위에 있어서  `layout_constraintXXX_XXX`가 필요없습니다.

sjjeong commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/data/source/UpbitDataSource.kt:
```diff
@@ -0,0 +1,7 @@
+package dev.daeyeon.gaasproject.data.source
+
+interface UpbitDataSource {
+    fun getTicker(success: (list: List<Ticker>) -> Unit?, fail:(msg: String) -> Unit?)
```
Unit 뒤에는 ?가 없어도 됩니다.

sjjeong commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/data/source/UpbitRepository.kt:
```diff
@@ -0,0 +1,56 @@
+
+class UpbitRepository : UpbitDataSource {
+
+    var markets: String? = null
```
```kotlin
var markets = ""
```
으로 하고 다른부분에
```kotlin
markets.isNullOrEmpty()로 처리하면 NonNull하게 사용할 수 있습니다.
```

sjjeong commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/TickerAdapter.kt:
```diff
@@ -0,0 +1,46 @@
+     * 비우기
+     */
+    fun clearList() {
+        tickerList.clear()
```
addList에 NotifyDataSetChanged()가 있다면 clearList에도 있어야합니다.
```kotlin
tickerAdapter?.clearList()
tickerAdapter?.addList(it)
```
이렇게 쓰는게 목적이라면 replaceList, resetList 등등으로 내부에 clear와 addAll을 같이해주는 함수를 따로 만들어주세요.

sjjeong commented on improve777/GAASProject/app/src/main/java/dev/daeyeon/gaasproject/network/NetworkManager.kt:
```diff
@@ -0,0 +1,27 @@
+import retrofit2.converter.gson.GsonConverterFactory
+
+object NetworkManager {
+    var instance: UpbitApi? = null
```
get할 때 초기화 해주고 싶다면 lazy 패턴을 사용하세요. NonNull하게 사용할 수 있습니다.

ddaeyeonkim committed:
.gitIgnore 추가

- codeStyleConfig.xml
- app/libs/

ddaeyeonkim committed:
멘토 피드백 반영하여 코드 개선

- UpbitRepository : markets notnull 타입으로 변경
- UpbitDataSource : Unit notnull 타입으로 변경
- TickerAdapter   : replaceList() 메서드 추가
- NetworkManager  : instance lazy 패턴 적용, createUpbitApi() 메소드 생성
- fragment_ticker.xml : recyclerView 제약조건 제거
- 잘못된 패키지명 수정

sjjeong commented:
Mentor Review Needed는 팀원들이 approve 해주고 해야합니다.

sjjeong reviewed (commented):
47839fdecfc463317acc5e0189aa76849811c117
확인했습니다.

sabgilhun reviewed (dismissed):
LGTM ㅎㅎ

sjjeong commented:
@sabgilhun approve 하셨으면 mentor review needed로 라벨 변경해주셔야합니다.

sjjeong commented:
@GongSulMin 리뷰해주세요.

GongSulMin reviewed (dismissed):
몰라서 찾아본
LGTM...

ddaeyeonkim committed:
통신 응답 null, empty 처리

ddaeyeonkim committed:
anko 라이브러리 추가 및 toast 수정
